caddytls/certificates.go

```diff
@@ -195,6 +208,8 @@ func fillCertFromLeaf(cert *Certificate, tlsCert tls.Certificate) error {
 		return err
 	}
 
+	cert.Leaf = leaf
```

Comment:
🤔 I realize you reference the Leaf cert elsewhere, but I'm wondering if setting this field has any other side-effects or implications? Or was it a mistake that we didn't do this before?